sdk/identity/azure-identity/azure/identity/_internal/interactive.py

```diff
@@ -187,7 +189,9 @@ def _acquire_token_silent(self, *scopes, **kwargs):
                     continue
 
                 now = int(time.time())
-                result = app.acquire_token_silent_with_error(list(scopes), account=account, **kwargs)
+                result = app.acquire_token_silent_with_error(
+                    list(scopes), account=account, claims_challenge=kwargs.get("claims")
+                )
```

Comment:
The removal of `**kwargs` will make acquiring SSH certificate stop working as `data` is passed in `kwargs`. 

Azure CLI is aiming to support acquiring SSH certificate for Service Principal as well (https://github.com/Azure/azure-sdk-for-python/pull/16397).

Reply:
Summarizing offline discussion here, there are two orthogonal concerns: acquiring SSH certificates, and allowing applications to pass keyword arguments through get_token to MSAL. Neither is supported today. Acquiring SSH certificates may be supported in a future version. Passing MSAL arguments through get_token is possible in some cases but really isn't supportable because routine maintenance requires internal changes (e.g. #16449) that can break applications relying on this behavior. My goal with this change is to prevent applications from taking a dependency on such implementation details.